memcache/memcache.go

```diff
@@ -403,6 +430,9 @@ func (c *Client) dial(addr net.Addr) (net.Conn, error) {
 	}
 
 	if ne, ok := err.(net.Error); ok && ne.Timeout() {
+		// In the case of a timeout, we might need to reestablish a connection to one or more servers, for example in case of an IP change
+		// Fire this off in a goroutine so it doesn't delay a response back to the client
+		c.backgroundReconnect()
```

Comment:
It would be nice to test this

Reply:
Done [e764d08](https://github.com/grafana/gomemcache/pull/18/commits/e764d08ea2ede32d1d22550aaa14f81232b744a2)